File: Backend/user-service/spring_boot_backend_template/spring_boot_backend_template/src/main/java/com/backend/entity/Bus.java
package com.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "buses")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Bus {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String busName;     // Volvo, Neeta

    @Column(nullable = false, unique = true)
    private String busNumber;   // MH12AB1234

    @Column(nullable = false)
    private String busType;     // AC, NON_AC, SLEEPER

    @Column(nullable = false)
    private int totalSeats;     // 40, 45
}

File: Backend/admin-service/admin/admin-service/Models/Bus.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace admin_service.Models
{
    [Table("buses")]
    public class Bus
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long Id { get; set; }

        [Required]
        [Column("bus_name")]
        public string BusName { get; set; }  // Volvo, Neeta

        [Required]
        [Column("bus_number")]
        public string BusNumber { get; set; }  // MH12AB1234

        [Required]
        [Column("bus_type")]
        public string BusType { get; set; }  // AC, NON_AC, SLEEPER

        [Required]
        [Column("total_seats")]
        public int TotalSeats { get; set; }  // 40, 45
    }
}
